core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1515,4 +1594,63 @@ private RemoveSnapshots removeSnapshots(Table table) {
     RemoveSnapshots removeSnapshots = (RemoveSnapshots) table.expireSnapshots();
     return (RemoveSnapshots) removeSnapshots.withIncrementalCleanup(incrementalCleanup);
   }
+
+  private StatisticsFile writeStatsFileForCurrentSnapshot(Table table, File statsLocation)
```

Comment:
I don't think there's a reason to pass table to this method. I think this should accept a String location, a FileIO, and a snapshot ID.

This should also not use `File` for writing.

Reply:
This was taken from existing code.
https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestRemoveOrphanFilesAction.java#L917

I will modify the current testcase.